src/main/java/com/daylong/songr/AlbumSongs.java

```diff
@@ -11,24 +11,52 @@
     long id;
 
     @ManyToOne
-    Album album;
+    Album songParent;
 
-    public String songName;
-    public int songDuration;
+    private String songName;
+    private double songDuration;
+    private int trackNumber;
+
+    public AlbumSongs(Long id, String songName, double songDuration, int trackNumber, Album songParent){
+        this.id = id;
+        this.songName = songName;
+        this.songDuration = songDuration;
+        this.trackNumber = trackNumber;
+        this.songParent = songParent;
```

Comment:
Get rid of id and songParent. Those are handled by Jpa. If you manually change them you will be sad.

Reply:
Good call.

Though I am used to being sad....